File: WarehouseBooking/WarehouseBooking.API/Models/Booking.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WarehouseBooking.API.Models
{
    public enum BookingStatus
    {
        Pending,
        Confirmed,
        Active,
        Completed,
        Cancelled
    }

    public class Booking
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        public Guid UserId { get; set; }

        [Required]
        public Guid WarehouseUnitId { get; set; }

        [Required]
        public DateTime StartDate { get; set; }

        [Required]
        public DateTime EndDate { get; set; }

        [Required]
        public BookingStatus Status { get; set; } = BookingStatus.Pending;

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal TotalAmount { get; set; }

        [MaxLength(1000)]
        public string? Notes { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        [ForeignKey("UserId")]
        public User User { get; set; } = null!;

        [ForeignKey("WarehouseUnitId")]
        public WarehouseUnit WarehouseUnit { get; set; } = null!;

        public ICollection<Payment> Payments { get; set; } = new List<Payment>();
    }
}

File: WarehouseBooking/warehouse-booking-web/src/types/booking.ts
export enum BookingStatus {
  Pending = 0,
  Confirmed = 1,
  Cancelled = 2,
  Completed = 3
}

export interface BookingDto {
  id: string;
  userId: string;
  warehouseUnitId: string;
  startDate: string;
  endDate: string;
  totalAmount: number;
  status: BookingStatus;
  createdAt: string;
  warehouseUnit?: {
    unitNumber: string;
    squareMeters: number;
    warehouse?: {
      name: string;
      location: string;
      address: string;
    };
  };
}

export interface CreateBookingRequest {
  warehouseUnitId: string;
  startDate: string;
  endDate: string;
  totalAmount: number;
}
